mlos_bench/mlos_bench/services/config_persistence.py

```diff
@@ -454,29 +425,31 @@ def load_services(self, json_file_names: Iterable[str],
             service.register(self.build_service(config, global_config, service).export())
         return service
 
-    def load_tunables(self, json_file_names: Iterable[str],
-                      parent: Optional[TunableGroups] = None) -> TunableGroups:
+    def _load_tunables(self, json_file_names: Iterable[str],
+                       parent: TunableGroups) -> TunableGroups:
         """
-        Load a collection of tunable parameters from JSON files.
+        Load a collection of tunable parameters from JSON files into the parent
+        TunableGroup.
+
+        This helps allow standalone environment configs to reference
+        overlapping tunable groups configs but still allow combining them into
+        a single instance that each environment can reference.
 
         Parameters
         ----------
         json_file_names : list of str
             A list of JSON files to load.
         parent : TunableGroups
-            An optional collection of tunables to add to the new collection.
+            A (possibly empty) collection of tunables to add to the new collection.
 
         Returns
         -------
         tunables : TunableGroup
-            Create a new collection of tunable parameters.
+            The larger collection of tunable parameters.
         """
         _LOG.info("Load tunables: '%s'", json_file_names)
-        groups = TunableGroups()
-        if parent is not None:
-            groups.update(parent)
         for fname in json_file_names:
             config = self.load_config(fname)
             assert isinstance(config, dict)
-            groups.update(TunableGroups(config))
-        return groups
+            parent.merge(TunableGroups(config))
+        return parent
```

Comment:
This is part of the key change: we put the child tunables into the parent and return the parent.
The child tunables are then ephemeral and only used to create the object when loading the config from disk.